### PR TITLE
Correctly handle updates to WebExtensions without an guid

### PR DESCRIPTION
**Pull request by EnTeQuAk**
I hope this fixes it once and for all.

### CODE REVIEW

File: src/olympia/files/utils.py [hunk 1]
```diff
@@ -507,6 +507,9 @@ def check_xpi_info(xpi_info, addon=None):
 
     guid_optional = xpi_info.get('is_webextension', False)
 
+    if addon and not guid and guid_optional:
+        xpi_info['guid'] = guid = addon.guid
```
Comment (diox) [inline src/olympia/files/utils.py]: Add a comment maybe ? This is starting to be confusing, I understand it now, I'm not sure I will in 6 months :)

File: src/olympia/files/tests/test_models.py [hunk 2]
```diff
@@ -423,6 +423,20 @@ def test_guid_dupe_webextension_guid_given(self):
             self.parse(filename='webextension.xpi')
         assert e.exception.messages == ['Duplicate add-on ID found.']
 
+    def test_guid_nomatch_webextension(self):
+        addon = Addon.objects.create(guid='xxx', type=1)
```
Comment (diox) [inline src/olympia/files/tests/test_models.py]: Can you use a valid guid just in case ? Same for below.

File: src/olympia/files/utils.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/files/utils.py]: absolutely :)

File: src/olympia/files/tests/test_models.py (hunk 2 above)
Comment (EnTeQuAk) [inline src/olympia/files/tests/test_models.py]: Fixed in 12f7ca3

File: src/olympia/files/utils.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/files/utils.py]: done in 12f7ca3

**Comment by EnTeQuAk** [comment]
@diox mind a re-r?

**Comment by diox** [comment]
r+

**Pull request merged**